### PR TITLE
merge adjacent roads with the same text for labeling

### PR DESCRIPTION
**Pull request by ansis**
ported from https://github.com/mapbox/mapbox-gl-js/pull/683

@kkaefer can you review?

### CODE REVIEW

File: src/mbgl/util/merge_lines.hpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        auto left = rightIndex.find(leftKey);
+        auto right = leftIndex.find(rightKey);
+
+        if ((left != rightIndex.end()) && (right != leftIndex.end()) && (left->second != right->second)) {
+            // found lines with the same text adjacent to both ends of the current line, merge all three
+            unsigned int j = mergeFromLeft(features, leftIndex, leftKey, right, geometry);
+            unsigned int i = mergeFromRight(features, rightIndex, left, rightKey, features[j].geometry);
+
+            leftIndex.erase(leftIndex.find(leftKey));
```
Comment (jfirebaugh) [inline src/mbgl/util/merge_lines.hpp]: `leftIndex.erase(leftKey)` / `rightIndex.erase(rightKey)`, so you don't have to think about whether `find` might return an `end()` iterator (which is invalid for `erase`).

**Comment by jfirebaugh** [comment]
Can you port the tests too please?

File: src/mbgl/util/merge_lines.hpp (hunk 1 above)
Comment (ansis) [inline src/mbgl/util/merge_lines.hpp]: fixed in b54c0e2

**Comment by ansis** [comment]
tests ported in 4fbb7d6

**Pull request merged**